activesupport/test/message_verifier_test.rb

```diff
@@ -99,6 +99,19 @@ class MessageVerifierMetadataTest < ActiveSupport::TestCase
     @verifier = ActiveSupport::MessageVerifier.new("Hey, I'm a secret!", verifier_options)
   end
 
+  def test_verify_method_raises_invalid_signature
```

Comment:
This should be 2 tests: `test_verify_raises_when_purpose_differs` and `test_verify_raises_when_expired`.